### PR TITLE
[Task] use fetch and avatar

### PR DESCRIPTION
**Pull request by ranjabi**
## Story/Task

_Fix use fetch for wrong post address and use transparent avatar_

## Summary

-   _Add oskm-logo.svg, rename previous oskm logo to oskm-logo-text._
-   _Add config in tailwind_
- _Add base layout to wrong post address_

I tried to rethrow the promise from ghostApi.tsx but it leads to uncaught promise from ghost api which is unclear because the try catch block already declared in useFetch. I think the try catch block in useFetch.tsx is not needed because the returned value from url promise is whether fulfilled/rejected.

## Test Plan

-   [x] _Visit wrong post address_
-   [x] _Visit wrong page ex: ../abou_

## Screenshot

image
image
image

### CODE REVIEW

File: src/service/ghostAPI.ts [hunk 1]
```diff
@@ -63,9 +62,8 @@ export const fetchSinglePost = async (postId: string) => {
         }
         return data;
     } catch (err: any) {
-        errMessage = err.message || 'An error occured from server';
+        return err;
```
Comment (IloveNooodles) [inline src/service/ghostAPI.ts:65]: ini udah di cek ga erro?

File: src/pages/ErrorPage.tsx [hunk 2]
```diff
@@ -1,18 +1,21 @@
...
-        <div className="min-h-screen flex items-center justify-center flex-col text-center bg-gradient-to-b from-Orange to-LightOrange">
-            <p className="m-0 mb-2 font-Heading text-h3">
-                {message || 'Internal server error'}{' '}
-            </p>
-            <p className="bg-Yellow rounded p-2 hover:shadow-sm">
-                Please refresh the page
-            </p>
-        </div>
-    </Animate>
+    <BaseLayout>
```
Comment (IloveNooodles) [inline src/pages/ErrorPage.tsx:7]: ini coba sesuaiin lgg dari wiwid harusnya baselatoutnya digapuss

File: src/hooks/useFetch.tsx [hunk 3]
```diff
@@ -10,22 +10,20 @@ const useFetch = (url: Promise<any>, page?: number) => {
...
-            if (data) {
-                setFetchedData({
-                    data: data.results || data,
-                    isLoading: false,
-                    error: false,
-                    message: '',
-                });
-            }
-        } catch (err: any) {
+        const data = await url;
```
Comment (IloveNooodles) [inline src/hooks/useFetch.tsx:13]: kayaknya tetep kita try catch deh iniii

Comment (IloveNooodles) [reply src/hooks/useFetch.tsx:13]: soalnya kalo ada error di datanya tetep harus di catch? apakah sudah di cek smua kasusnya make data instanceof errornya?

File: src/service/ghostAPI.ts (hunk 1 above)
Comment (ranjabi) [reply src/service/ghostAPI.ts:65]: engga, aman

File: src/hooks/useFetch.tsx (hunk 3 above)
Comment (ranjabi) [reply src/hooks/useFetch.tsx:13]: imo datanya udh dicatch di ghostAPI.ts jadinya promise di file ini gabakal nerima throw error? idk blm dapet alternatif lain sih. kasusnya so far domain salah/connection_refused kek gambar di bawah bakal kehandle. tambahin catch lagi berarti ngulang kode yg di dalem else?
image

File: src/service/ghostAPI.ts (hunk 1 above)
Comment (IloveNooodles) [reply src/service/ghostAPI.ts:65]: ini nanti error yang di returnnya apa yakk btw

Comment (IloveNooodles) [reply src/service/ghostAPI.ts:65]: kita dapet error messagenya dari si use fetchnya ya berarti

**Pull request merged**